### DISCUSSION TITLE
Streamsync component ID customization and access

### DISCUSSION THREAD

**Discussion by ZBMO** [writer/writer-framework#170] [Q&A]
several questions

1. Is it possible to access the calling component's ID from its event handler?
2. is it possible to set a custom ID on a non-custom component?
3. is it possible to make events from child components 'bubble-up' to a parent component's event handler? without handling it manually through many event handler functions that point to one?

Thank you, we are trying to make many inputs create the same type of event (for web sockets) without having to create 100 different event handler functions.

**Comment by FabienArcellier** [comment]
Hello,

1) I think it's the same question than #169. I don't think there is yet a way to manage this case. Maybe @ramedina86 will have a tip to do that.

If not, I will try to design a change to add the `target_components` available in context of event handler.

2) I don't think it's possible. There is no general id mechanism to manage custom id on the frontend
